util/datastructures.py

```diff
@@ -7,21 +10,61 @@ def __init__(self):
         self.nodes = []
 
     def __init__(self, nodes, edges):
```

Comment:
Do we want typing here?

Reply:
Yes, definitely. I wasn't sure back then what types they will have.